Source/Scene/ImageryLayer.js

```diff
@@ -866,6 +971,81 @@ define([
         imagery.state = ImageryState.TEXTURE_LOADED;
     };
 
+    /**
+     * Creates WebGL textures for each image in a given multi-source {@link Imagery} instance.
+     * @private
+     *
+     * @param {Context} context The rendered context to use to create textures.
+     * @param {Imagery} imagery The imagery for which to create a texture.
+     * @param
+     */
+    ImageryLayer.prototype._createMultipleTextures = function(context, imagery) {
+        var imageryProvider = this._imageryProvider;
+        var projectedImages = imagery.projectedImages;
+
+        // If this imagery provider has a discard policy, use it to check if this
+        // image should be discarded.
+        if (defined(imageryProvider.tileDiscardPolicy)) {
+            var discardPolicy = imageryProvider.tileDiscardPolicy;
+            if (defined(discardPolicy)) {
+                // If the discard policy is not ready yet, transition back to the
+                // RECEIVED state and we'll try again next time.
+                if (!discardPolicy.isReady()) {
+                    imagery.state = ImageryState.RECEIVED;
+                    return;
+                }
+
+                // Mark discarded imagery tiles invalid.  Parent imagery will be used instead.
+                if (discardPolicy.shouldDiscardImage(projectedImages[0])) {
```

Comment:
Could this have a mix of images that should be discarded and some that should not? If so, you should discard the ones that need to be and, if all are discarded, set the state to invalid.

Reply:
> discard the ones that need to be

If only some of the images are discarded do we need to fill in the missing areas using the parent image? Is it too heavy-handed to mark the entire tile as invalid if one of its images is discarded?

If we're filling in, maybe we could grab the parent's reprojected image and copy over select areas. We can use the computed texcoords like a mask but read out of the parent texture using the fragment coordinates and another uniform or something.

Reply:
> grab the parent's reprojected image

There might be complications with this too, if the parent isn't available would we want to go for the nearest ancestor instead? If so, would we want to regenerate the tile when the parent becomes available?

-___-

Reply:
Those are good questions. Maybe open an issue? What do you think @lilleyse?

Reply:
Had a discussion offline, we'll just discard the entire "target" tile for now. I'll document the behavior.